daemon/containerd/image_delete.go

```diff
@@ -59,28 +61,278 @@ func (i *ImageService) ImageDelete(ctx context.Context, imageRef string, force,
 		return nil, err
 	}
 
+	imgID := image.ID(img.Target.Digest)
+
+	if isImageIDPrefix(imgID.String(), imageRef) {
+		return i.deleteAll(ctx, img, force, prune)
+	}
+
+	singleRef, err := i.isSingleReference(ctx, img)
+	if err != nil {
+		return nil, err
+	}
+	if !singleRef {
+		err := i.client.ImageService().Delete(ctx, img.Name)
+		if err != nil {
+			return nil, err
+		}
+		i.LogImageEvent(imgID.String(), imgID.String(), "untag")
```

Comment:
Yeah! I wondered the same (but did this over a while so I forgot to call it out when opening the PR) but yes, I left it like this since this was the non-containerd implementation as well. I'll leave it for a follow up to change both/discuss any potential impacts of doing so.